suzieq/poller/controller/source/base_source.py

```diff
@@ -40,10 +42,10 @@ def __init__(self, input_data) -> None:
             'jump_host_key_file',
             'ignore_known_hosts'
         ]
-        errors = self._validate_config(input_data)
-        if errors:
-            raise InventorySourceError('Inventory validation failed: {}'
-                                       .format(errors))
+
+        self._validate_config(input_data)
+        self._validate_device()
```

Comment:
Do we also want to complain about keywords we don't know? This can happen because of a typo for example.

Reply:
I added validation for all fields which are not known for namespaces, devices, auths and sources. If an argument is not declared. an error will be shown

Reply:
Excellent